### DISCUSSION TITLE
Blocage IP aussi

### DISCUSSION THREAD

**Discussion by glitchim** [bokub/conso-api#48] [Aide] [answered]
Bonjour,

J'utilise depuis un moment ton API et je la trouve super pratique, tellement simple en 2 appels curl par jour c'est du bonheur ! Tellement simple que j'ai procrastiné sur mon script et laissé un retry après 20 minutes pour les rares fois ou les données n'étaient pas disponible.

Sauf que ce week-end, mon authorisation chez Enedis a expiré (je pense) et mon script c'est emballé, et relancé le lendemain, doublant encore le nombre de requêtes ... Bref, la boulette classique.

J'ai **totalement désactivé** le retry, je suis **désolé** pour ces ~250 (?) requêtes en 3 jours.

Peux-tu débloquer mon IP ? Avec Cloudflare Ray ID: 8718e3883cd49e7f tu as les infos ? Merci d'avance !
Et merci pour cette API surtout !!

Si je peux me permettre une suggestion, peut-être qu'un blocage temporaire des IP fautives t'éviterait de gérer des messages comme le mien ... Si par exemple mon blocage durait 1 semaine et que je le savais via la documentation, je ne viendrais pas ici demander un déblocage (et créer un compte GitHub pour ça) ... Vive l'automatisation ;)

**Comment by glitchim** [comment]
Bon si j'en crois https://github.com/bokub/conso-api/commit/dcfeae77461a90689472e8d487df98115f50b369 c'est peine perdue 😢 ?? Pas le droit à l'erreur ? Dur ...

**Reply by bokub** [reply, marked as answer]
> qu'un blocage temporaire des IP fautives ... Si par exemple mon blocage durait 1 semaine et que je le savais via la documentation

Alors non c'est pas si simple, parce que la plupart des IP bloquées continuent à spammer de requêtes pendant des semaines voire des mois sans même sans rendre compte (quitte à pas faire attention, autant faire les choses jusqu'au bout !)

Pour que tu rendes compte du volume, c'est entre 15 000 et 20 000 requêtes bloquées par jour par les IPs qui se sont faites blacklister, dont certaines depuis 2023.

Y'a aussi ceux qui demandent à être débloqué mais qui n'ont même pas analysé le problème, et disent ne pas comprendre ce qui s'est passé. Même si je cède à leur demande, je perds du temps pour rien vu qu'ils seront re-bloqués quelques jours plus tard....

___
Allez t'as du bol je suis de bonne humeur aujourd'hui

**Reply by glitchim** [reply]
Merci beaucoup !
Effectivement je n'ai pas d'info sur les volumes que tu gères ...
